src/app/calculator/steam/stack-loss/stack-loss-form/stack-loss-by-volume/stack-loss-by-volume.component.html

```diff
@@ -118,4 +119,16 @@
       </div>
     </div> -->
   </div>
-</form>
\ No newline at end of file
+</form>
+
+<div bsModal #materialModal="bs-modal" class="modal fade" tabindex="-1" role="dialog" aria-labelledby="materialModalLabel"
+  aria-hidden="true" [config]="{backdrop: 'static'}">
+  <div class="modal-dialog modal-lg">
+    <div class="modal-content">
+      <div>
+        <app-flue-gas-material *ngIf="showModal" (closeModal)="hideMaterialModal($event)" (hideModal)="hideMaterialModal($event)"
```

Comment:
instead of showModal use `this.materialModal.isShown` and then you can remove the "showModal" from the component.

Reply:
Same goes for other form